File: Application/projeto-site/models/Setor.js
'use strict';

/* 
lista e explicação dos Datatypes:
https://codewithhugo.com/sequelize-data-types-a-practical-guide/
*/

module.exports = (sequelize, DataTypes) => {
    let Setor = sequelize.define('Setor',{
		id: {
			field: 'idSetor',
			type: DataTypes.INTEGER,
			primaryKey: true,
			autoIncrement: true
		},		
		nome: {
			field: 'nome',
			type: DataTypes.STRING,
			allowNull: false
		},
		qtdSensores: {
			field: 'qtdSensores',
			type: DataTypes.INTEGER,
			allowNull: false
		},
		fkFilial: {
			field: 'fkFilial',
			type: DataTypes.INTEGER,
            allowNull: false,
			references: {
                model: 'Filial',
                key: 'idFilial',
			}
		},
	}, 
	{
		tableName: 'Setor', 
		freezeTableName: true, 
		underscored: true,
		timestamps: false,
	});

    return Setor;
};

File: Application/cadastro-componentes/src/main/java/model/bean/Setor.java
package model.bean;

public class Setor {
    
    private int idSetor;
    private String nome;
    private int fkFilial;
    private int qtdSensores;

    public int getIdSetor() {
        return idSetor;
    }

    public String getNome() {
        return nome;
    }

    public int getFkFilial() {
        return fkFilial;
    }

    public void setIdSetor(int idSetor) {
        this.idSetor = idSetor;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public void setFkFilial(int idFilial) {
        this.fkFilial = idFilial;
    }

    public void setQtdSensores(int qtdSensores) {
        this.qtdSensores = qtdSensores;
    }

    public int getQtdSensores() {
        return qtdSensores;
    }
    
    
    
}
